src/modules/core/components/OmniPicker/AdditionalOmniPicker.tsx

```diff
@@ -0,0 +1,277 @@
+      | KeyboardEvent<HTMLElement>,
+  ) => void;
+
+  /** Is called when a data entry was picked */
+  onPick?: (itemData: OmniPickerData) => void;
+
+  /** Title for the dropdown header */
+  title?: string | MessageDescriptor;
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  choose: Choose;
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  close?: (
+    evt:
+      | MouseEvent
+      | SyntheticEvent<HTMLInputElement>
+      | KeyboardEvent<HTMLElement>,
+  ) => void;
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  getItem: (data: OmniPickerData[], selectedIdx: number) => OmniPickerData;
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  filteredData: OmniPickerData[];
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  id: string;
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  inputRef: HTMLElement | null;
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  keyUsed: boolean;
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  select: (idx: number) => void;
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  selected: number;
+
+  /** @ignore Will be injected by `withOmniPicker` */
+  triggerElement: HTMLElement | null;
+}
+
+class AdditionalOmniPicker extends Component<Props> {
+  elm?: HTMLElement | null;
+
+  static displayName = 'OmniPicker';
+
+  static defaultProps = {
+    appearance: { position: 'bottom' },
+    renderEmpty: () => <OmniPickerItemEmpty />,
+  };
+
+  componentDidMount() {
+    if (document.body) {
+      document.body.addEventListener(
+        'click',
+        this.handleOutsideClick as any,
+        true,
+      );
+    }
+  }
+
+  componentWillUnmount() {
+    if (document.body) {
+      document.body.removeEventListener(
+        'click',
+        this.handleOutsideClick as any,
+        true,
+      );
+    }
+  }
+
+  handleOpen = () => {
+    const { onOpen } = this.props;
+    if (typeof onOpen == 'function') {
+      onOpen();
+    }
+  };
+
+  handleClose = () => {
+    const { onClose } = this.props;
+    if (typeof onClose == 'function') {
+      onClose();
+    }
+  };
+
+  handleReset = () => {
+    const { onReset } = this.props;
+    if (typeof onReset == 'function') {
+      onReset();
+    }
+  };
+
+  handleKeyUp = (evt: KeyboardEvent<HTMLElement>) => {
+    const { onKeyUp } = this.props;
+    if (typeof onKeyUp == 'function') {
+      onKeyUp(evt);
+    }
+  };
+
+  handleKeyDown = (evt: KeyboardEvent<HTMLElement>) => {
+    const { onKeyDown } = this.props;
+    if (typeof onKeyDown == 'function') {
+      onKeyDown(evt);
+    }
+  };
+
+  handleChange = (evt: SyntheticEvent<HTMLInputElement>) => {
+    const { onChange } = this.props;
+    if (typeof onChange == 'function') {
+      onChange(evt);
+    }
+  };
+
+  handleFocus = (evt: SyntheticEvent<HTMLInputElement>) => {
+    const { onFocus } = this.props;
+    if (typeof onFocus == 'function') {
+      onFocus(evt);
+    }
+  };
```

Comment:
Just a tough though, I know these were copy - pasted from the parent component, but I think theses can actually be refactored as a hook.

Something like:
```tsx
// The hook
enum Handle {
  OPEN = 'open',
  CLOSE = 'close',
  RESET = 'reset',
  ...
}

interface Handlers {
  [key: string]?: (evt: SyntheticEvent<HTMLInputElement>) => void;
}

const useHandleFactory = <D>(handles: Handle[], dependencies: D): Handlers => {
  const allHandlers: Handlers = {};
  const [props] = dependencies;

  handles.map((handle: Handle) => {
    const callbackName = `on${handle.charAt(0).toUpperCase()}${handle.slice(1)}`;
    const handlerName = `handle${callbackName.charAt(0).toUpperCase()}${callbackName.slice(1)}`;
    allHandlers[handlerName] = (evt: SyntheticEvent<HTMLInputElement>) => {
      const handlerCallback = props[callbackName];
      if (typeof handlerCallback == 'function') {
        handlerCallback(evt);
      }
    };
  });
};

// Then you would call it something like this (inside the component)

const { handleOnOpen, handleOnClose, handleOnReset, ... } = useHandleFactory(
  [Handle.OPEN, Handle.CLOSE, Handle.RESET, ... ],
  [this.props],
);
```

Of course, this is all theoretic only, I don't expect you to spend the time to actually implement this.